### PR TITLE
Revise knitr Lesson from Feedback

### PR DESCRIPTION
**Pull request by brymz**
As per #546.

### CODE REVIEW

File: materials/knitr-examp.Rmd [hunk 1]
```diff
@@ -19,8 +19,8 @@ library(scales)
 Data is from the [Portal project teaching database](http://figshare.com/articles/Portal_Project_Teaching_Database/1314459) published on Figshare. We need the surveys table for our analysis:
 
 ```{r, cache=TRUE}
-# Setting cache=TRUE lets you re-use the results of this code chunk instead of
-# re-calculating or re-downloading it from scratch.
+# Setting cache=TRUE lets you re-use the results of this code chunk in other code
+# chunks instead of re-calculating or re-downloading it each time.
```
Comment (ethanwhite) [inline materials/knitr-examp.Rmd]: I'm not sure this change exactly right. Code chunks share state automatically. I'm pretty sure that `cache` is primarily operating between knits, not within a single knit.

**Comment by brymz** [comment]
> the demo in a separate file...just happened because last year I happened to commit the Rmd.

The `Rmd` was committed in Aug with the lecture notes: https://github.com/datacarpentry/semester-biology/commit/7e7ca05b587694ab198165cf611f8cbf0e44cf0b

> I write the Rmd file live in front of the class... [from] code stored in the main `MD` document.

Thanks for the clarification. I definitely support live-coding. I will go ahead and move the `Rmd` code chucks to the `MD` and remove the `Rmd`. Are there any specific 'talking points' or stages of development that you would like me to include in the revision?

**Comment by ethanwhite** [comment]
> Are there any specific 'talking points' or stages of development that you would like me to include in the revision?

There is stuff that should be added, but I don't have time to think through it at the moment. Feel free to add something yourself or I'll take care of it later (you can open an issue and assign it to me).

**Comment by brymz** [comment]
As per our discussion, I have integrated the `Rmd` code into the lecture notes `MD` and added talking points. I have included a mention of the R Notebooks to address #557. The lesson can be easily shifted to a Markdown or Notebook focus from there.

I am not happy with the way the code blocks are rendered when using the block spacing method as was used in previous versions of this file. The code blocks are misaligned and there is extra spacing that could get in the way if the live-coder decides to copy/paste the code blocks. I am going to use the background html tags `<pre><code>` to set the blocks 'manually'. I committed the block spacing option in case we decided we preferred that version.

**Comment by brymz** [comment]
Ready to review/merge.

**Comment by ethanwhite** [comment]
This all looks great. Did we end up dropping Dave's `Rpres` material entirely? If so it might be worth adding some of it back to the end demonstrating the ability to make slides using knitr.

**Comment by brymz** [comment]
I made sure the content was merged into the lecture notes. I will add the directions to make an `Rpres`.

**Comment by brymz** [comment]
`Rpres` section added.

**Comment by ethanwhite** [comment]
Thanks @brymz!

**Pull request merged**